src/Player.Vm.Api/Infrastructure/BackgroundServices/CallbackBackgroundService.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Carnegie Mellon University. All Rights Reserved.
+// Released under a MIT (SEI)-style license. See LICENSE.md in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Player.Api.Client;
+using Player.Vm.Api.Data;
+using Player.Vm.Api.Domain.Services;
+using Player.Vm.Api.Features.Vms;
+using Player.Vm.Api.Infrastructure.Options;
+
+namespace Player.Vm.Api.Infrastructure.BackgroundServices
+{
+    public interface ICallbackBackgroundService
+    {
+        void AddEvent(WebhookEvent e);
+        Task<VmMap[]> CloneMaps(ViewCreated form, CancellationToken ct);
+    }
+
+    public class CallbackBackgroundService : ICallbackBackgroundService
+    {
+        private ActionBlock<WebhookEvent> _eventQueue;
+        private readonly IServiceScopeFactory _scopeFactory;
+        private readonly IMapper _mapper;
+        private readonly ILogger<CallbackBackgroundService> _logger;
+        private readonly IHttpClientFactory _clientFactory;
+
+        public CallbackBackgroundService(IServiceScopeFactory scopeFactory, IMapper mapper, ILogger<CallbackBackgroundService> logger, IHttpClientFactory clientFactory)
+        {
+            _scopeFactory = scopeFactory;
+            _mapper = mapper;
+            _logger = logger;
+            _clientFactory = clientFactory;
+
+            _eventQueue = new ActionBlock<WebhookEvent>(async e => {
+                switch (e.Name)
+                {
+                    case "View Created":
+                        // Since payload field is of type object, the json serialization in player makes it into string containing json
+                        var payload = JsonConvert.DeserializeObject<ViewCreated>(e.Payload.ToString());
+                        await CloneMaps(payload, new CancellationToken());
+                        break;
+                    case "View Deleted":
+                        throw new NotImplementedException();
```

Comment:
We should implement this to cleanup after a View is deleted. Remove any cloned maps I guess?